### PR TITLE
Expose recaptcha reply (and score) to the controller

### PR DESCRIPTION
**Pull request by Intrepidd**
As mentioned in #343

It would be very helpful to be able to get the recaptcha v3 score, for logging purposes for example, or to have different responses given the score (more than just a binary yes/no)

This PR is WIP, I just want to validate the implementation, as I'm adding some public methods and tweaking a bit the `verify_via_api_call` by returning a tuple which is a bit weird.

If you are ok with the implem, I will add tests and update the doc.
Otherwise, if you have suggestions I'm all ears

Thanks

### CODE REVIEW

File: lib/recaptcha.rb [hunk 1]
```diff
@@ -65,10 +65,12 @@ def self.verify_via_api_call(response, options)
     verify_hash['remoteip'] = options[:remote_ip] if options.key?(:remote_ip)
 
     reply = api_verification(verify_hash, timeout: options[:timeout])
-    reply['success'].to_s == 'true' &&
+    success = reply['success'].to_s == 'true' &&
       hostname_valid?(reply['hostname'], options[:hostname]) &&
       action_valid?(reply['action'], options[:action]) &&
       score_above_threshold?(reply['score'], options[:minimum_score])
+
+    return success, reply
```
Comment (esale) [inline lib/recaptcha.rb]: `return [success, reply]`

Comment (grosser) [inline lib/recaptcha.rb]: how about not breaking anyones existing code ? :)
```suggestion
    if options[:reply]
      [success, reply]
    else
      success
    end
```

Comment (esale) [reply lib/recaptcha.rb]: Developer's wisdom :)

Comment (Intrepidd) [reply lib/recaptcha.rb]: I find my version simpler to read, why do you suggest this more verbose approach ?

Comment (Intrepidd) [reply lib/recaptcha.rb]: Nice thanks. How about something a little more explicit like `include_reply` or `with_reply` ?

Comment (esale) [reply lib/recaptcha.rb]: You must return an array. But in any case, @grosser already gave us a better solution.

Comment (Intrepidd) [reply lib/recaptcha.rb]: `return success, reply` is exactly the same as `return [success,reply]` :) 

Such as calling `foo(a: 42)` is the same as calling `foo({a: 42})`, just a style difference / preference

Comment (grosser) [reply lib/recaptcha.rb]: either is fine 👍

Comment (Intrepidd) [reply lib/recaptcha.rb]: Fab, if the rest is ok with you I'll work on adding tests and doc tomorrow

File: lib/recaptcha/adapters/controller_methods.rb [hunk 2]
```diff
@@ -24,7 +24,8 @@ def verify_recaptcha(options = {})
               options = options.merge(remote_ip: remoteip.to_s) if remoteip
             end
 
-            Recaptcha.verify_via_api_call(recaptcha_response, options)
+            success, @_recaptcha_reply = Recaptcha.verify_via_api_call(recaptcha_response, options)
```
Comment (grosser) [inline lib/recaptcha/adapters/controller_methods.rb]: this will have to pass in `with_reply: true` or something like that

Comment (grosser) [reply lib/recaptcha/adapters/controller_methods.rb]: ah nvm ... `a, b = true` works out fine ... so the user can decide if they want the reply

File: lib/recaptcha.rb (hunk 1 above)
Comment (grosser) [reply lib/recaptcha.rb]: yeah, just needs test + docs :)

**Comment by Intrepidd** [comment]
Alright @grosser, let me know if this is missing anything !

**Comment by grosser** [comment]
5.5.0

**Comment by Intrepidd** [comment]
Awesome thanks !

**Pull request merged**